lib/model/settings.dart

```diff
@@ -203,6 +203,55 @@ enum BoolGlobalSetting {
   };
 }
 
+/// An int-valued, account-independent setting the user might set.
+///
+/// These are recorded in the table [IntGlobalSettings].
+/// To read the value of one of these settings, use [GlobalSettingsStore.getInt];
+/// to set the value, use [GlobalSettingsStore.setInt].
+///
+/// To introduce a new setting, add a value to this enum.
+/// Avoid re-using any old names found in the "former settings" list.
+///
+/// To remove a setting, comment it out and move to the "former settings" list.
+/// Tracking the names of settings that formerly existed is important because
+/// they may still appear in users' databases, which means that if we were to
+/// accidentally reuse one for an unrelated new setting then users would
+/// unwittingly get those values applied to the new setting,
+/// which could cause very confusing buggy behavior.
+///
+/// (If the list of former settings gets long, we could do a migration to clear
+/// them from existing installs, and then drop the list.  We don't do that
+/// eagerly each time, to avoid creating a new schema version each time we
+/// finish an experimental feature.)
+enum IntGlobalSetting {
+  /// A non-setting to ensure this enum has at least one value.
+  ///
+  /// (This is also handy to use in tests.)
+  placeholderIgnore,
+
+  /// A pseudo-setting recording the id of the account the user has visited most
+  /// recently, from the list of all the available accounts on the device.
+  ///
+  /// In some cases, this may point to an account that doesn't actually exist on
+  /// the device, for example, when the last visited account is logged out and
+  /// another account is not visited during the same running session. For cases
+  /// like these, it's the responsibility of the code that reads this value to
+  /// check for the availability of the account that corresponds to this id.
+  lastVisitedAccountId,
```

Comment:
Now that you have a `placeholderIgnore` value in this enum, I think it should be easy to split this commit:
24f4e44c9 settings: Make general int global settings, so as to add without migrations

up into two commits:
* one with the existing commit message, which adds all the general infrastructure but not `lastVisitedAccountId` specifically;
* one that adds `lastVisitedAccountId`, along with the GlobalStore methods for it and the nav observer that keeps it updated.

I guess the cost of that is that then there'd be two migrations, and two added copies of the schema. Still, I think it'd make the commit a lot clearer to read. So let's go ahead and do that.